google-cloud-notification/src/main/java/com/google/cloud/notification/Notification.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.notification;
+
+import com.google.api.core.BetaApi;
+import com.google.cloud.Service;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.util.List;
+
+/**
+ * An interface for notifications in Google Cloud Storage.
+ *
+ * @see <a href="https://cloud.google.com/storage/docs">Google Cloud Storage</a>
+ */
+public interface Notification extends Service<StorageOptions> {
```

Comment:
I don't think this is the right relationship between `Notification` and `StorageOptions`.

Reply:
Notifications uses Storage as the underlying RPC caller. I think it'd be redundant to create a new NotificationsOptions class that mimics StorageOptions.

Reply:
But then should `Notification` be a "Service" at all? Why not make it clear that it is a syntactic shell around other service options, and have the user pass in the `Storage` object to us?

Reply:
Ok, Notification is no longer a service, and it explicitly accepts a StorageOptions as a constructor parameter.

Reply:
Why not just accept a `Storage` object? By passing in `StorageOptions`, you might get multiple `Storage` clients generated.

Reply:
The StorageOptions is required for retrySettings, etc. The Storage client object can't handle NotificationInfo, which contains TopicName, which is a Pubsub entity.

Reply:
If you have a Storage object, you can get StorageOptions out of it. From StorageOptions, you can call getRpc() and cast it to StorageRpc, and make calls from there.

Reply:
Ok, Notification client now takes in a Storage object as constructor parameter. PTAL